PluginGenerator/JarManifestBuilder.cs

```diff
@@ -0,0 +1,162 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="JarManifestBuilder.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace SonarQube.Plugins
+{
+    /// <summary>
+    /// Builds a valid v1.0 Jar Manifest.
+    /// See http://docs.oracle.com/javase/6/docs/technotes/guides/jar/jar.html#JAR%20Manifest
+    /// </summary>
+    public class JarManifestBuilder
+    {
+        /// <summary>
+        /// Magic string that must appear first in the file
+        /// </summary>
+        private const string ManifestVersionEntry = "Manifest-Version: 1.0";
+
+        /// <summary>
+        /// Required manifest file name
+        /// </summary>
+        private const string ManifestFileName = "MANIFEST.MF";
+
+        private const string ValidNamePattern = "^[0-9a-zA-Z_-]+$";
+        private const int MaxNameLength = 70;
+        private const int MaxLineLength = 72;
+
+        private readonly IDictionary<string, string> properties = new Dictionary<string, string>();
+
+        #region Public methods
+
+        /// <summary>
+        /// Sets the value of a property in the manifest file.
+        /// Any existing value will be overwritten.
+        /// </summary>
+        public void SetProperty(string name, string value)
+        {
+            CheckAttributeNameIsValid(name);
+
+            this.properties[name] = value;
+        }
+
+        public bool TryGetValue(string name, out string value)
+        {
+            return this.properties.TryGetValue(name, out value);
+        }
+
+        /// <summary>
+        /// Builds and returns the manifest
+        /// </summary>
+        public string GetManifest()
```

Comment:
BuildManifest() might be a better name, GetX suggests a simple fetch / query operation